scripts/api-extractor.ts

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable import/no-extraneous-dependencies */
+
+// eslint-disable-next-line no-restricted-imports
+import {
+  resolve as resolvePath,
+  relative as relativePath,
+  dirname,
+} from 'path';
+import fs from 'fs-extra';
+import {
+  Extractor,
+  ExtractorConfig,
+  CompilerState,
+  ExtractorLogLevel,
+} from '@microsoft/api-extractor';
+import { ApiPackage, ApiModel } from '@microsoft/api-extractor-model';
+import { MarkdownDocumenter } from '@microsoft/api-documenter/lib/documenters/MarkdownDocumenter';
+
+const tmpDir = resolvePath(__dirname, '../node_modules/.cache/api-extractor');
+
+/**
+ * Yup
+ */
+const {
+  PackageJsonLookup,
+} = require('@rushstack/node-core-library/lib/PackageJsonLookup');
+
+const old = PackageJsonLookup.prototype.tryGetPackageJsonFilePathFor;
+PackageJsonLookup.prototype.tryGetPackageJsonFilePathFor = function tryGetPackageJsonFilePathForPatch(
+  path: string,
+) {
+  if (
+    path.includes('@material-ui') &&
+    !dirname(path).endsWith('@material-ui')
+  ) {
+    return undefined;
+  }
+  return old.call(this, path);
+};
+
+const DOCUMENTED_PACKAGES = [
+  'packages/backend-common',
+  'packages/catalog-client',
+  'packages/catalog-model',
+  'packages/cli-common',
+  'packages/config',
+  'packages/config-loader',
+  // 'packages/core',
+  // 'packages/core-api',
```

Comment:
There would be more entries than just these two :p

I think allowlist makes sense in this case tbh, instead of having a list of example apps, storybook, etc. If we add plugins to the mix we could flip it around tho.